testdata/p4_16_bmv_errors/emit-struct-bmv2.p4

```diff
@@ -0,0 +1,98 @@
+/*
+    Copyright 2018 MNK Consulting, LLC.
+    http://mnkcg.com
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+//  This test fails for bmv2 backend because the backend does not support
```

Comment:
Shouldn't this test fail for all backends?
Do we want to support emit of raw structs?

Reply:
The P4-16 specification allows emit of raw struct.  I can look into both p4test and ebpf backends to see support for such an emit.

Reply:
When you say "raw struct", do you mean a struct containing only `bit<W>` fields, for example?  If so, where in the P4-16 specification is that allowed?  What I have found is that emit on a struct has the meaning of emit on each of its members, recursively, and that "It is illegal to invoke emit on an expression of whose type is a base type, enum, or error" (Section 15.1 "Data insertion into packets", version 1.1 of P4_16 language spec).

Reply:
If that is true you should instead file an issue. This test should probably be filed with the issue instead; you don't want to test for a bug that does not belong to your PR, so perhaps you can adapt or delete this test.

Reply:
if @jafingerhut is right, then this test should fail for all back-ends; it probably should fail in the front-end. So it is not a very useful test.

Reply:
I understand a raw struct as a struct with only bit-fields.  I do know an error is not serializable.   A SerializableEnum can be supported in a struct.   Section 15.1 does not allow base types with emit, but base types in a Boolean and Integer inside a struct should be allowed because the types are serializable.  Such a struct should be legal with emit.

I added this test, using a raw struct, only for the bmv2 backend in p4_16_bmv_errors directory, because this backend fails for a raw struct.  The test also serves as a reminder to complete the bmv2 backend for such an emit.   I haven't tested p4test and ebpf backends.  Depending upon what passes or fails, I can remove/keep the test and see if any frontend checks need to be added to code.

Reply:
This test seems to have no connection to any of your changes, so probably it should be deleted, including the reference files it generates.

Reply:
The emit statement will fail with or without your changes, so the failure you are testing is not related to your new changes.

Reply:
ok, will do.

Reply:
I agree that it might be very reasonable to change the P4_16 language spec to support emit'ing of structs containing only things like `bit<W>`, `int<W>`, and serializable enums.  I am simply pointing out that this is a case where the spec doesn't _yet_ say this is legal.  If we are OK with the implementation leading the spec here, that doesn't bother me, as long as we realize it _is_ leading the spec.

Reply:
Andy, I agree with you.  
Thanks!